doc/modules/naive_bayes.rst

```diff
@@ -259,6 +259,109 @@ categories for each feature :math:`i` are represented with numbers
 :math:`0, ..., n_i - 1` where :math:`n_i` is the number of available categories
 of feature :math:`i`.
 
+.. _general_naive_bayes:
+
+General Naive Bayes
+-------------------
+
+:class:`GeneralNB` implements multiple naive Bayes models across the
+features in the dataset by assuming different distributions for different
+features, while maintaining conditional independence between every pair of
+features given the value of a class variable.
+
+A practical use for this metaestimator is when encountering data with
+both numerical and categorical features. For example, suppose our data
+had 5 features where the first three are numerical and the rest categorical.
+We then proceed to assume that numerical features follow the Gaussian
+distribution and the categorical features follow the categorical
+distribution, i.e.,
+
+.. math::
+
+   X_1 \mid y \sim \text{Normal}(\mu_1,\sigma_1^2) \\
+   X_2 \mid y \sim \text{Normal}(\mu_2,\sigma_2^2) \\
+   X_3 \mid y \sim \text{Normal}(\mu_3,\sigma_3^2) \\
+   X_4 \mid y \sim \text{Categorical}(\textbf{p}_4) \\
+   X_5 \mid y \sim \text{Categorical}(\textbf{p}_5)
+
+Let's see how we `GeneralNB` is used with this toy dataset. We first import
+the libraries and prepare the data:
+
+   >>> import numpy as np
+   >>> import pandas as pd
+   >>> from sklearn.naive_bayes import GeneralNB, GaussianNB, CategoricalNB
+   >>>
+   >>> X = np.array([[1.5, 2.3, 5.7, 0, 1],
+   ...               [2.7, 3.8, 2.3, 1, 0],
+   ...               [1.7, 0.1, 4.5, 1, 0]])
+   >>> y = np.array([1, 0, 0])
+   >>> X_test = np.array([[1.5, 2.3, 5.7, 0, 1]])
+
+In the `GeneralNB` constructor,
+define a name (for easy access of the fitted estimators later)
+and the corresponding columns for every naive Bayes model.
+Below we defined two tuples, one for the `GaussianNB()` and
+one for the `CategoricalNB()` model.
+This manner of specification is similar to that of *transformers* in
+:class:`ColumnTransformer <sklearn.compose.ColumnTransformer>` or
+*pipeline* in :class:`Pipeline <sklearn.pipeline.Pipeline>`.
```

Comment:
I think you mean `steps`, not `pipeline`. But I'd rather you just drop the reference to Pipeline here: the syntax is much more like ColumnTransformer

Reply:
Okay.